### PR TITLE
Add engine_getPayloadV4 and engine_newPayloadV4 for Electra

### LINKED ISSUE #8084: Add/Update EngineGetPayloadV4 to support exits
execution API PR: https://github.com/ethereum/execution-apis/pull/528

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
As per https://github.com/ethereum/execution-apis/pull/528 and https://github.com/ethereum/execution-apis/pull/531

## Fixed Issue(s)

- related to a task in #7898 
- fixes #8084 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/DepositReceiptV1.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  @JsonDeserialize(using = BytesDeserializer.class)
+  public final Bytes signature;
+
+  @JsonSerialize(using = UInt64AsHexSerializer.class)
+  @JsonDeserialize(using = UInt64AsHexDeserializer.class)
+  public final UInt64 index;
+
+  public DepositReceiptV1(
+      @JsonProperty("pubkey") final Bytes48 pubkey,
+      @JsonProperty("withdrawal_credentials") final Bytes32 withdrawalCredentials,
```
Comment (lucassaldanha) [inline ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/DepositReceiptV1.java]: I believe this is supposed to be `withdrawalCredentials`

**Review by lucassaldanha — COMMENTED** [review]
PR is looking good. However I believe one of the fields has the wrong name (using snake case instead of came case).

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/DepositReceiptV1.java (hunk 1 above)
Comment (StefanBratanov) [reply ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/DepositReceiptV1.java]: You are right. Good catch!

**Review by lucassaldanha — APPROVED** [review]
LGTM :)

**Pull request merged**